### PR TITLE
Feature booking controller cud

### PR DESCRIPTION
**Pull request by Parashchuk**

### CODE REVIEW

File: src/main/java/lux/fm/bookingservice/validation/BookingDatesValidator.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    public boolean isValid(Object object, ConstraintValidatorContext constraintValidatorContext) {
+        Object checkInDate = new BeanWrapperImpl(object).getPropertyValue(checkIn);
+        Object checkOutDate = new BeanWrapperImpl(object).getPropertyValue(checkOut);
+        if(checkInDate == null) {
+            return true;
+        }
+
+        if(checkOutDate == null) {
+            return true;
+        }
```
Comment (RedArtelerist) [inline src/main/java/lux/fm/bookingservice/validation/BookingDatesValidator.java]: remove this

File: src/main/java/lux/fm/bookingservice/service/impl/BookingServiceImpl.java [hunk 2]
```diff
@@ -36,12 +40,38 @@ public List<BookingResponseDto> findMyBookings(String username) {
...
+    }
+
+    private void validateDates(LocalDate checkIn, LocalDate checkOut, Booking booking) {
+        if((checkIn != null && booking.getCheckOut().isBefore(checkIn))
+                || (checkOut != null && booking.getCheckIn().isAfter(checkOut))
+        ) {
+            throw new BookingInvalidDateException(
+                    "CheckOut date can't be earlier than checkIn date"
+            );
+        }
```
Comment (RedArtelerist) [inline src/main/java/lux/fm/bookingservice/service/impl/BookingServiceImpl.java]: To delete

File: src/main/java/lux/fm/bookingservice/model/Booking.java [hunk 3]
```diff
@@ -15,11 +15,13 @@
 import java.time.LocalTime;
 import lombok.Getter;
 import lombok.Setter;
+import org.hibernate.annotations.Where;
 
 @Getter
 @Setter
 @Entity
 @Table(name = "bookings")
+@Where(clause = "status != 'CANCELED'")
```
Comment (RedArtelerist) [inline src/main/java/lux/fm/bookingservice/model/Booking.java:24]: Add Sql delete set Canceled

**Review by RedArtelerist — CHANGES_REQUESTED** [review]
Great, left some comments

**Review by viacheslav-torbin — APPROVED** [review]
LGTM 🔥

**Pull request merged**